src/Compilers/Core/Portable/InternalUtilities/SetWithInsertionOrder.cs

```diff
@@ -57,5 +78,7 @@ public IEnumerator<T> GetEnumerator()
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
         public ImmutableArray<T> AsImmutable() => _elements.ToImmutableArrayOrEmpty();
+
+        public T this[int i] => _elements[i];
```

Comment:
ArrayBuilder doesn't (and can't) have a `ref` returning indexer, so this can't be.